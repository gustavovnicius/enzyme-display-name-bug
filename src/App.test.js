import React from 'react';
import { shallow } from 'enzyme';
import { connect } from 'react-redux';

it('renders welcome message', () => {
  const mapStateToProps = () => ({});
  const Component = () => <div />;

  Component.displayName = 'MyComponent';

  const Connected = connect(mapStateToProps)(Component);

  const Wrapper = () => (
    <div>
      <Connected />
    </div>
  );

  const wrapper = shallow(<Wrapper />);

  expect(wrapper).toMatchSnapshot();
});

it('render component without correct displayName', () => {
  const mapStateToProps = () => ({});
  const Component = (body) => <div>{body}</div>;
  const Body = () => <div>Hello there</div>;

  Component.displayName = 'MyComponent';
  Body.displayName = 'Body';

  const MemoBody = React.memo(Body);

  const Connected = connect(mapStateToProps)(Component);
  const ConnectedBody = connect(mapStateToProps)(MemoBody)

  const Wrapper = () => (
    <div>
      <Connected body={<ConnectedBody />} />
    </div>
  );

  const wrapper = shallow(<Wrapper />);

  expect(wrapper).toMatchSnapshot();
});
