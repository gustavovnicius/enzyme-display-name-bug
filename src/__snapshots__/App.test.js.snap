// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component without correct displayName 1`] = `
<div>
  <Connect(MyComponent)
    body={<Memo(ConnectFunction) />}
  />
</div>
`;

exports[`renders welcome message 1`] = `
<div>
  <Connect(MyComponent) />
</div>
`;
